snippets/csharp/System.Net.Sockets/Socket/Overview/socket.cs

```diff
@@ -3,97 +3,103 @@
   * This example creates a socket connection to the server specified by the user,
   * using port 80. Once the connection has been established it asks the server for
   * the content of its home page. If no server name is passed as argument to this
-  * program, it sends the request to the current machine.
+  * program, it sends the request to example.com.
   * */
- //<Snippet1>
 using System;
 using System.Text;
-using System.IO;
-using System.Net;
 using System.Net.Sockets;
+using System.Threading.Tasks;
+using System.Threading;
+using System.Linq;
 
 public class GetSocket
 {
-    private static Socket ConnectSocket(string server, int port)
+    public static async Task Main(string[] args)
     {
-        Socket s = null;
-        IPHostEntry hostEntry = null;
-
-        // Get host related information.
-        hostEntry = Dns.GetHostEntry(server);
-
-        // Loop through the AddressList to obtain the supported AddressFamily. This is to avoid
-        // an exception that occurs when the host IP Address is not compatible with the address family
-        // (typical in the IPv6 case).
-        foreach(IPAddress address in hostEntry.AddressList)
-        {
-            IPEndPoint ipe = new IPEndPoint(address, port);
-            Socket tempSocket =
-                new Socket(ipe.AddressFamily, SocketType.Stream, ProtocolType.Tcp);
-
-            tempSocket.Connect(ipe);
-
-            if(tempSocket.Connected)
-            {
-                s = tempSocket;
-                break;
-            }
-            else
-            {
-                continue;
-            }
-        }
-        return s;
+        Uri? uri = args.Any() ? new Uri(args[0]) : null;
+
+        // Sync:
+        SendHttpRequest(uri);
+
+        // Async:
+        await SendHttpRequestAsync(uri);
     }
 
-    // This method requests the home page content for the specified server.
-    private static string SocketSendReceive(string server, int port)
+//<Snippet1>
+private static void SendHttpRequest(Uri? uri = null, int port = 80)
+{
+    uri ??= new Uri("http://example.com");
+
+    // Construct a minimalistic HTTP/1.1 request
+    byte[] requestBytes = Encoding.ASCII.GetBytes(@$"GET {uri.AbsoluteUri} HTTP/1.1
```

Comment:
```suggestion
    byte[] requestBytes = Encoding.ASCII.GetBytes(@$"GET {uri.AbsoluteUri} HTTP/1.0
```

In Kestrel at least, this will prevent chunked responses which should lead to easier-to-read output.